src/api/metadataApi.ts

```diff
@@ -6,11 +6,50 @@
  */
 
 import explorerServiceAxios from './explorerServiceAxios';
+import helpers from '../utils/helpers';
+import { METADATA_RETRY_LIMIT, DOWNLOAD_METADATA_RETRY_INTERVAL } from '../constants';
 
 const metadataApi = {
-  async getDagMetadata(id: string, network: string) {
-    const axios = await explorerServiceAxios(network);
-    return axios.get(`metadata/dag`, { params: { id }});
+  async getDagMetadata(id: string, network: string, options = {}) {
+    type optionsType = {
+      retries: number,
+      retryInterval: number,
+    };
+    const newOptions: optionsType = Object.assign({
+      retries: METADATA_RETRY_LIMIT,
+      retryInterval: DOWNLOAD_METADATA_RETRY_INTERVAL,
+    }, options);
+
+    let { retries, retryInterval } = newOptions;
+    while (retries >= 0) {
+      const axios = await explorerServiceAxios(network);
+      try {
+        const response = await axios.get(`metadata/dag`, { params: { id }});
+        if (response.data) {
+          return response.data;
+        } else {
+          // Error downloading metadata
+          // throw Error and the catch will handle it
+          throw Error;
```

Comment:
Should we customize the error? We can also add a message to it but I feel a customized error is better.

Reply:
Done https://github.com/HathorNetwork/hathor-wallet-lib/pull/266/commits/1c957a38e489e17807245a45df8f97105841e99d